packages/components/Callout/macos/Callout.swift

```diff
@@ -321,4 +378,30 @@ private class FlippedVisualEffectView: NSVisualEffectView {
 	}
 }
 
+
+/// NSEvent localMonitors are an _old_ API that requires you to do some  memory management yourself.
+/// You _must_ call `removeMonitor`exactly once per monitor created from. This helper class enforces
+/// that with the use of an extra boolean to track whether we have already allocated/released our monitor.
+/// https://developer.apple.com/documentation/appkit/nsevent/1533709-removemonitor
+private class GuardedEventMonitor: Any {
+	
+	private var isMonitorAllocated = false
+	private var allocatedMonitor: Any?
+	
+	
+	func addLocalMonitorForEvents(matching mask: NSEvent.EventTypeMask, handler block: @escaping (NSEvent) -> NSEvent?) {
+		if (!isMonitorAllocated) {
+			allocatedMonitor = NSEvent.addLocalMonitorForEvents(matching: mask, handler: block)
+			isMonitorAllocated = true
+		}
+	}
+	
+	func removeMonitor() {
```

Comment:
do we need to call this removeMonitor in deinit() as well?

Reply:
In one of my implementations while debugging, I had a version of that. Theoretically it should be safe with my guarding. I can try it out and see what happens.

Reply:
with the zombie or memory leak profiling should help unittest it?

Reply:
The zombie tool is telling me that no `guardedEventMonitors` are sticking around after I close a menu. Now that `removeMonitor` is called in deinit, I'm pretty confident no underlying NSEventMonitors are sticking around either.